xcom2-launcher/xcom2-launcher/Forms/SettingsDialog.cs

```diff
@@ -117,8 +117,14 @@ private void bOK_Click(object sender, EventArgs e)
             var modFoldersChanged = !Settings.ModPaths.OrderBy(x => x).SequenceEqual(newModPaths.OrderBy(x => x));
             IsRestartRequired = sentryEnableChanged || gamePathChanged || modFoldersChanged;
 
+            // Verify settings
+            if (!Directory.Exists(gamePathTextBox.Text)) {
+                MessageBox.Show("The specified base path does not exist.", "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+                return;
+            }
+            
             // Apply changes
-            Settings.GamePath = Path.GetFullPath(gamePathTextBox.Text);
+            Settings.GamePath = gamePathTextBox.Text;
```

Comment:
Do we not need `Path.GetFullPath`?

Reply:
I do not see how this would make sense. Even IF a user would specify an incomplete path, GetFullPath() uses the "current directory" to create a fully qualified path. This normally isn't the game directory. So no, I think it is not required. At least I do not see a valid use case.

Reply:
Oh and we are catching incomplete paths with Directory.Exists() anyway.

Reply:
From `Directory.Exists()` docs:

> The path parameter is permitted to specify relative or absolute path information. Relative path information is interpreted as relative to the current working directory.

So it will accept relative paths.. I think we should keep `Path.GetFullPath()` just to be sure

Reply:
Valid point. As Directory.Exists() already "catches" invalid characters or an empty path, leaving GetFullPath() in should be fine.